File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    queue = deque(zip(priorities, range(len(priorities))))
    count = 1
    
    while queue:
        top = queue.popleft()
        
        if any (top[0] < item[0] for item in queue):
            queue.append(top)
        elif top[1] == location:
            return count
        else:
            count += 1
        
    return count
            

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    const arr = priorities.map((priority, index) => [priority, index]);
    let count = 1, start = 0
    
    function shift() {
        const result = arr[start].slice()
        
        arr[start++][0] = 0
        
        return result
    }
    
    while (arr) {
        const top = shift();
        if (arr.some(item => item[0] > top[0])) {
            arr.push(top)
        } else if (top[1] === location) {
            return count
        } else {
            count += 1
        }
    }
    
    
    return count
}